packages/ckeditor5-link/src/linkui.js

```diff
@@ -154,7 +155,21 @@ export default class LinkUI extends Plugin {
 
 		// Execute link command after clicking the "Save" button.
 		this.listenTo( formView, 'submit', () => {
-			editor.execute( 'link', formView.urlInputView.fieldView.element.value, formView.getDecoratorSwitchesState() );
+			const { value } = formView.urlInputView.fieldView.element;
+
+			// It's good to check the config for the second time on submit.
+			// This will cover more dynamic cases, when protocol might change after the element has been rendered.
+			const defaultProtocol = editor.config.get( 'link.defaultProtocol' );
+
+			// The regex checks for the protocol syntax ('xxxx://' or 'xxxx:')
+			// or non-word charecters at the begining of the link ('/', '#' etc.).
+			const isProtocolNeeded = !!defaultProtocol && !( /^((\w+:(\/{2,})?)|(\W))/gmi ).test( value );
+			const isEmail = ( /[\w-]+@[\w-]+\.+[\w-]+/gmi ).test( value );
+
+			const protocol = isEmail && 'mailto:' || defaultProtocol;
```

Comment:
```suggestion
			const protocol = isEmail ? 'mailto:' : defaultProtocol;
```

Straight forward if-else requires less time to parse ;)

Reply:
🙈

Reply:
I don't know what happened here 😬